sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ProcessSessionMessageEventArgs.cs

```diff
@@ -108,11 +116,36 @@ public class ProcessSessionMessageEventArgs : EventArgs
         {
             Message = message;
             _manager = manager;
+            // for mocking
+            if (_manager != null)
+            {
+                _manager.SessionLockRenewed += SessionLockRenewed;
+                _manager.SessionLockLost += SessionLockLost;
+            }
 
             // manager would be null in scenarios where customers are using the public constructor for testing purposes.
             _sessionReceiver = (ServiceBusSessionReceiver) _manager?.Receiver;
             _receiveActions = new ProcessorReceiveActions(message, manager, false);
             CancellationToken = cancellationToken;
+            LockExpiryCancellationToken = _receiveActions.GetLockExpiryCancellationToken(message);
+        }
+
+        private void SessionLockLost(object sender, EventArgs e)
+        {
+            foreach (var messageAndTokenSource in Messages)
+            {
+                messageAndTokenSource.Value.Cancel();
+            }
+        }
+
+        private void SessionLockRenewed(object sender, EventArgs e)
+        {
+            var utcNow = DateTimeOffset.UtcNow;
+            foreach (var messageAndTokenSource in Messages)
+            {
+                messageAndTokenSource.Key.LockedUntil = _sessionReceiver.SessionLockedUntil;
```

Comment:
Is my assumption correct that we should refresh the messages with the session receiver lock until?

Reply:
No, the LockedUntil property is not updated for session messages. Even for non-session messages, the property update would happen inside the RenewMessageLock method.

Reply:
Ok in that sense my implementation would be breaking. I'll revert